File: application-bemyplan/src/main/java/com/deploy/bemyplan/controller/auth/dto/response/LoginResponse.java
package com.deploy.bemyplan.controller.auth.dto.response;

import lombok.*;

@ToString
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class LoginResponse {

    private String sessionId;

    private Long userId;

    private String nickname;

    public static LoginResponse of(String sessionId, Long userId, String nickname) {
        return new LoginResponse(sessionId, userId, nickname);
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/deploy/bemyplan/auth/adapter/in/api/LoginResponse.kt
package com.deploy.bemyplan.auth.adapter.`in`.api

data class LoginResponse(
    val token: String,
    val sessionId: String,
    val userId: Long,
    val nickname: String,
) {
    companion object {
        fun of(token: String, sessionId: String, userId: Long, nickname: String): com.deploy.bemyplan.auth.adapter.`in`.api.LoginResponse {
            return com.deploy.bemyplan.auth.adapter.`in`.api.LoginResponse(token, sessionId, userId, nickname)
        }
    }
}